lib/mongoid/reloadable.rb

```diff
@@ -17,16 +17,14 @@ module Reloadable
     #
     # @return [ Document ] The document, reloaded.
     def reload
-      if @atomic_selector
-        # Clear atomic_selector cache for sharded clusters. MONGOID-5076
-        remove_instance_variable('@atomic_selector')
```

Comment:
I am not sure I understand why removing `@atomic_selector` is not necessary now.

Reply:
I should have left a note here about that. Thanks for mentioning it. The reason is that it is dead code -- it was an artifact of a fix from the 7.x series, and is no longer needed. I removed it as part of some casual cleaning up here.